ako-gateway-api/nodes/avi_model_l7_translator.go

```diff
@@ -50,32 +59,89 @@ func (o *AviObjectGraph) BuildChildVS(key string, routeModel RouteModel, parentN
 	childNode.Tenant = lib.GetTenant()
 	childNode.EVHParent = false
 
-	_, _, parentName := lib.ExtractTypeNameNamespace(parentNsName)
+	//_, _, parentName := lib.ExtractTypeNameNamespace(parentNsName)
 	childNode.ServiceMetadata = lib.ServiceMetadataObj{
 		Gateway: parentName,
 	}
 	childNode.ApplicationProfile = utils.DEFAULT_L7_APP_PROFILE //TODO: move this to the routeModel
 	childNode.ServiceEngineGroup = lib.GetSEGName()
 	childNode.VrfContext = lib.GetVrf()
-	childNode.VHDomainNames = routeModel.ParseRouteRules().Hosts
+	//childNode.VHDomainNames = routeModel.ParseRouteRules().Hosts
 
 	// TODO: add markers
 	//	evhNode.AviMarkers = lib.PopulateVSNodeMarkers(namespace, host, infraSettingName)
+	childNode.AviMarkers = utils.AviObjectMarkers{
+		GatewayName: parentName,
+		Namespace:   parentNs,
+	}
+	found, hosts := akogatewayapiobjects.GatewayApiLister().GetGatewayRouteToHostname(parentNs, parentName)
+	if found {
+		childNode.VHDomainNames = hosts
+		childNode.AviMarkers.Host = hosts
+	} else {
+		return
+	}
 
 	// create pg pool from the backend
-	o.BuildPGPool(key, childNode, routeModel, rule)
+	o.BuildPGPool(key, childNode, routeModel, parentNsName, rule, listeners)
 
 	// create vhmatch from the match
 	o.BuildVHMatch(key, childNode, routeModel, rule)
 
 	// create the httppolicyset if the filter is mentioned
 	o.BuildHTTPPolicySet(key, childNode, routeModel, rule)
+	parentNode[0].EvhNodes = append(parentNode[0].EvhNodes, childNode)
 }
 
-func (o *AviObjectGraph) BuildPGPool(key string, childVsNode *nodes.AviEvhVsNode, routeModel RouteModel, rule *Rule) {
-
-	// create the PG from backends
-
+func (o *AviObjectGraph) BuildPGPool(key string, childVsNode *nodes.AviEvhVsNode, routeModel RouteModel, parentNsName string, rule *Rule, listeners []string) {
+
+	parentSlice := strings.Split(parentNsName, "/")
+	parentNs, parentName := parentSlice[0], parentSlice[1]
+	listenerSlice := strings.Split(listeners[0], "/")
+	listenerProtocol := listenerSlice[2]
+	PGName := akogatewayapilib.GetPoolGroupName(parentNs, parentName,
+		routeModel.GetNamespace(), routeModel.GetName(),
+		utils.Stringify(rule.Matches))
+	PG := nodes.AviPoolGroupNode{
+		Name:   PGName,
+		Tenant: lib.GetTenant(),
+	}
+	for _, backend := range rule.Backends {
+		svcObj, err := utils.GetInformers().ServiceInformer.Lister().Services(backend.Namespace).Get(backend.Name)
+		if err != nil {
+			utils.AviLog.Debugf("key: %s, msg: there was an error in retrieving the service", key)
+			return
+		}
+		poolName := akogatewayapilib.GetPoolName(parentNs, parentName,
+			routeModel.GetNamespace(), routeModel.GetName(),
+			utils.Stringify(rule.Matches),
+			backend.Namespace, backend.Name, strconv.Itoa(int(backend.Port)))
+		poolNode := &nodes.AviPoolNode{
+			Name:     poolName,
+			Tenant:   lib.GetTenant(),
+			Protocol: listenerProtocol,
```

Comment:
This protocol must match the protocol in the service. Seems to me that something is wrong here.
@akshayhavile could you share your views?

Reply:
I had a discussion with Akshay. We have to set the health monitors according to the listener protocols. For the time being, the system-TCP is enough and in future, we may have to change the behavior in AKO to set the HM based on the application.